### ISSUE TITLE
Printing out custom children nodes of BaseNode 

### ISSUE THREAD

**Issue by kj3moraes** [kayjan/bigtree#63]
Hello, I want to have 2 children nodes of BaseNode called TextNode and HeaderNode. Both of the them are defined as follows:

```python

class TextNode(BaseNode):
   def __init__(self, text:str, **kwargs):
        self.text = text
        super().__init__(**kwargs)

class HeaderNode(BaseNode): 
  def __init__(self, header:str, headerNumber:int, **kwargs):
        self.header = header
        self.headerNumber = headerNumber
        self.name = header
        self.corpus = []
        super().__init__(**kwargs)
```

I want to be able to print these out using either `print_tree` or `yield_tree` but they only take in `Node` objects. Is there a way for you to extend the functionality of `print_tree` and `yeild_tree` to work with any children of BaseNode. 

For example, say I do something like this 

```python

class TextNode(BaseNode):
  ...
   def __str__(self):
      return self.text

class HeaderNode(BaseNode): 
  def __str__(self):
    return self.header
```

and then the two functions could call this instead of `_node.node_name` .

**Comment by kayjan** [comment]
Hi,

May I know why you want to extend from `BaseNode` instead of `Node`? Because `Node` is essentially `BaseNode` with a name input, which you can convert your `text`/`header` field to `node_name` field. You can have your own `text`/`header` field to take from `node_name` field if you really want those fields.

```python
from bigtree import Node

class TextNode(Node):
    def __init__(self, name: str, **kwargs):
        super().__init__(name, **kwargs)

    @property
    def text(self) -> str:
        return self.node_name
```

The reason why `print_tree` and `yield_tree` work with `Node` and not `BaseNode` is because to visualize a tree, we need minimally a node with name input, which we access with `node.node_name` instead of just using `print(node)` which you can defined with `__str__`.

Under the hood, `yield_tree` uses tree iterator `preorder_iter`, if you really want to implement with `BaseNode`, you can perform your own tree iteration and do a `print(node)`.

### TLDR
For fast fix, do extend from `Node` instead of `BaseNode`. Otherwise a longer fix would be to perform your own tree iteration.

**Comment by kj3moraes** [comment]
This solved it thanks.
